Source/Core/DolphinQt/Main.cpp

```diff
@@ -259,7 +307,8 @@ int main(int argc, char* argv[])
     Settings::Instance().ApplyStyle();
 
     MainWindow win{Core::System::GetInstance(), std::move(boot),
-                   static_cast<const char*>(options.get("movie"))};
+                   static_cast<const char*>(options.get("movie")), options.is_set("netplay_join"),
+                   netplay_game};
```

Comment:
```suggestion
    MainWindow win{Core::System::GetInstance(), std::move(boot),
                   static_cast<const char*>(options.get("movie"))};

    if (options.is_set("netplay_join"))
      win.NetPlayJoin();
    else if (netplay_game.has_value())
      win.NetPlayHost(*netplay_game);
```

Rather than altering the `MainWindow` constructor to have 3 mutually exclusive parameters I think it would be cleaner to make `NetPlayJoin` and  `NetPlayHost` public and do something like this.